core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -303,6 +303,11 @@ private DefaultBlockMaster(MetricsMaster metricsMaster, CoreMasterContext master
 
     MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_LOST_BLOCK_COUNT.getName(),
         this::getLostBlocksCount);
+    if (ServerConfiguration.global().getBoolean(
+        PropertyKey.MASTER_COUNT_TO_REMOVE_BLOCKS_ENABLE)) {
```

Comment:
Adding this metric makes sense to me. And I don't think you need to use a configuration to turn it on/off. If we control all metrics this way the configuration will be too much.

The performance is a concern, I get it. You can use `MetricsSysmtem.registerCachedGaugeIfAbsent()`. If the cached gauge update interval is not what you want, you should change the method to set the cache time.